[WIP] feat: add lambda auth docs (#3429)
aws-amplify/docs: docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md

> ## Configure Lambda Auth Mode with DataStore

Reviewer: ```suggestion
## Configure custom authorization logic with AWS Lambda
```

> You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you AppSync API, please see the Settings section of the AppSync console.

Reviewer: ```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you AppSync API, please see the Settings section of the AppSync console.
```

> You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.

Reviewer: ```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda function as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.
```

You will need to manage the details of token refreshes in your application code yourself. 

> When using Lambda authorizers with DataStore here's how you can specify a function for handling token refresh:

Reviewer: ```suggestion
Here's how you can specify a function for handling token refresh when using Lambda as an authorization mode with DataStore:
```
